plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -655,84 +672,91 @@ private ConnectorTableMetadata doGetTableMetadata(ConnectorSession session, Sche
-        getSerdeProperty(table, TEXT_FIELD_SEPARATOR_ESCAPE_KEY)
+        getSerdeProperty(tableParameterProvider, TEXT_FIELD_SEPARATOR_ESCAPE_KEY)
                 .ifPresent(fieldEscape -> properties.put(TEXTFILE_FIELD_SEPARATOR_ESCAPE, fieldEscape));
 
         // CSV specific properties
-        getCsvSerdeProperty(table, CSV_SEPARATOR_KEY)
+        getCsvSerdeProperty(tableParameterProvider, CSV_SEPARATOR_KEY)
                 .ifPresent(csvSeparator -> properties.put(CSV_SEPARATOR, csvSeparator));
-        getCsvSerdeProperty(table, CSV_QUOTE_KEY)
+        getCsvSerdeProperty(tableParameterProvider, CSV_QUOTE_KEY)
                 .ifPresent(csvQuote -> properties.put(CSV_QUOTE, csvQuote));
-        getCsvSerdeProperty(table, CSV_ESCAPE_KEY)
+        getCsvSerdeProperty(tableParameterProvider, CSV_ESCAPE_KEY)
                 .ifPresent(csvEscape -> properties.put(CSV_ESCAPE, csvEscape));
 
         // REGEX specific properties
-        getSerdeProperty(table, REGEX_KEY)
+        getSerdeProperty(tableParameterProvider, REGEX_KEY)
                 .ifPresent(regex -> properties.put(REGEX_PATTERN, regex));
-        getSerdeProperty(table, REGEX_CASE_SENSITIVE_KEY)
+        getSerdeProperty(tableParameterProvider, REGEX_CASE_SENSITIVE_KEY)
                 .ifPresent(regexCaseInsensitive -> properties.put(REGEX_CASE_INSENSITIVE, parseBoolean(regexCaseInsensitive)));
 
-        Optional<String> comment = Optional.ofNullable(table.getParameters().get(TABLE_COMMENT));
+        Optional<String> comment = Optional.ofNullable(tableParameterProvider.getParameter(TABLE_COMMENT));
 
-        String autoPurgeProperty = table.getParameters().get(AUTO_PURGE_KEY);
+        String autoPurgeProperty = tableParameterProvider.getParameter(AUTO_PURGE_KEY);
         if (parseBoolean(autoPurgeProperty)) {
             properties.put(AUTO_PURGE, true);
         }
 
         // Partition Projection specific properties
-        properties.putAll(partitionProjectionService.getPartitionProjectionTrinoTableProperties(table));
+        properties.putAll(partitionProjectionService.getPartitionProjectionTrinoTableProperties(tableParameterProvider));
+
+        Map<String, String> remainingProperties = tableParameterProvider.getUnconsumedProperty();
+        if (!remainingProperties.isEmpty()) {
+            properties.put(EXTRA_PROPERTIES, remainingProperties);
+        }
 
         return new ConnectorTableMetadata(tableName, columns.build(), properties.buildOrThrow(), comment);
     }
 
-    private static Optional<String> getCsvSerdeProperty(Table table, String key)
+    private static Optional<String> getCsvSerdeProperty(TableParameterProvider tableParameterProvider, String key)
     {
-        return getSerdeProperty(table, key).map(csvSerdeProperty -> csvSerdeProperty.substring(0, 1));
+        return getSerdeProperty(tableParameterProvider, key).map(csvSerdeProperty -> csvSerdeProperty.substring(0, 1));
```

Comment:
Are we certain than `csvSerdeProperty` is not empty? If it is we are having an exception here